chapter6_smartview/chapter6_smartview.md

```diff
@@ -0,0 +1,197 @@
+# Chapter 6 SmartView
+SmartView is an umbrella term for one of the live streaming technologies offered by Sitelink3D v2. Compared with MFK Live (Machine Forward Kinematics), SmartView involves less programming investment to start using and is capable of providing information at various resolutions and with configurable server side filtering. These features facilitate a set of live streaming capabilities that can be used to implement expressions that make sense to the end user.
+
+## SmartView Microservice
+The ```smart_view``` microservice manages requests for SmartView data. It streams data using SSE (server sent events) and manages the lifetimes of SmartApps discussed below.
+
+## SmartApps
+SmartView delivers functionality by means of SmartApps. A SmartApp is simply a piece of serverside JavaScript that provides information about a specific bounded context of site operations that a user may wish to subscribe to. Each SmartApp has a particular purpose and is only concerned with providing information within its domain. SmartApp examples include:
+
+- Machine locations and state.
+- Weight payloads sent between machines.
+- Haul activity.
+
+A SmartApp is identified by three settings: 
+
+- Key which is a string like ‘topcon/machines/machine_list’.
+- Version which is a UUID. 
+- Start time.
+
+To subscribe to a SmartApp, you will need to know the following settings:
+1. Site of interest: you need the site identifier (hereafter called SITE).
+2. SmartApp of interest (hereafter called APP_B64): SmartApps have two distinguishing parts of their identity as discussed above:
+ * Key.
+ * Version.
+3. Settings that relate to the SmartApp specifically.
+4. The server URL.
+5. An authorization token. See the token example [here](https://github.com/Sitelink3D-v2-Developer/sitelink3dv2-examples/blob/main/components/tokens/get_token.py) for more information.
+
+These get put together into a GET request like this: 
+
+1. SITE is left as a SHA-256 value. 
+2. APP_B64 is formed from the base 64 encoding of 
+ - ```key:version``` to start a specific version.
+ - ```key``` to start the most recent version.
+3. START_B64 is formed from the base 64 encoding of {"from":"continuous"} which provides a continuous stream of data.
+4. Any other settings passed as parameters.
+
+In addition, if your app has received values before, it can maintain a mapping {key}->{crc32-of-value} which, if sent
+to the server, will make the server pretend it has already sent values with those shas, meaning they will not be sent
+again.
```

Comment:
I think we should remove this paragraph for the time being. It needs tightening up (you haven't said exactly how the values are transmitted, and I think it's changing).

Reply:
Done.